### DISCUSSION TITLE
Optional parameters on custom functions

### DISCUSSION THREAD

**Discussion by Feterro07** [ezylang/EvalEx#389] [Ideas]
I see that the topic of missing parameters has been discussed before, however it seems reasonable to me the possibility of having optional parameters on a function. More specifically I have a function that performs some additional process depending on a flag parameter. It would be more user friendly if the flag was optional with a default value if it isn't sent, however this is not possible because a missing parameter is taken as an error. A new setting on the functionParameter tag could be a nice way of adding this option.

**Comment by uklimaschewski** [comment]
There is a way to achieve this, using the _vararg_ flag on the last parameter.
One example can be found in the (not yet released) _DT_FORMAT_ (_DateTimeFormatFunction_):
This function formats a date time value to a string and has an optional second _format_ parameter.

I personally like the Python way of optional keyword (named) parameters with default values.
But I am unsure, if this can be implemented without breaking some functionality.
